src/layout/node.ts

```diff
@@ -20,17 +22,23 @@ export default function(node: Node, source: Source): void {
         node = node.parentNode;
     }
 
-    let call = dom.has(node) ? "update" : "add";
+    let add = dom.has(node) === false;
+    let call = add ? "add" : "update";
     let parent = node.parentElement ? node.parentElement : null;
+    let insideFrame = node.ownerDocument !== document;
     switch (node.nodeType) {
-        case Node.DOCUMENT_NODE:
-            observe(node);
-            break;
         case Node.DOCUMENT_TYPE_NODE:
+            // Skip processing nodes within iframe if they were already removed from the DOM before we got to processing them.
+            // In those cases, we skip them and continue processing rest of the nodes.
+            let frame = insideFrame && node.parentNode ? dom.iframe(node.parentNode) : null;
+            if (insideFrame && frame === null) { break; }
             let doctype = node as DocumentType;
             let docAttributes = { name: doctype.name, publicId: doctype.publicId, systemId: doctype.systemId };
-            let docData = { tag: Constant.DOCUMENT_TAG, attributes: docAttributes };
-            dom[call](node, parent, docData, source);
+            let docData = { tag: insideFrame ? Constant.FRAME_DOCUMENT_TAG : Constant.DOCUMENT_TAG, attributes: docAttributes };
```

Comment:
What does this value represent?

Reply:
I've simplified this to not use it anymore. I'm using prefix like we already use with SVG. Thanks for pointing out.